src/main/scala/is/hail/expr/Relational.scala

```diff
@@ -1406,7 +1406,13 @@ case class MatrixMapCols(child: MatrixIR, newCol: IR, newKey: Option[IndexedSeq[
     val colValuesType = TArray(prev.typ.colType)
     val vaType = prev.typ.rvRowType
 
+    var initOpNeedsSA = false
+    var initOpNeedsGlobals = false
+
     val transformInitOp: (Int, IR) => IR = { (nAggs, initOpIR) =>
+      initOpNeedsSA = Mentions(initOpIR, "sa")
```

Comment:
Are you planning on including this in the other operations?

Reply:
I was not planning on it. My main work for this week is to finish the off heap stuff.

Reply:
ok.